File: Algorithms/Strings/Palindrome Index.py
import os
#
# Complete the 'palindromeIndex' function below.
#
# The function is expected to return an INTEGER.
# The function accepts STRING s as parameter.
#

def palindromeIndex(s):
    # Write your code here
    for i in range(len(s)//2):
        if s[i] != s[len(s)-1-i]:
            if s[i+1:len(s)-i] == s[i+1:len(s)-i][::-1]:
                return i
            elif s[i:len(s)-i-1] == s[i:len(s)-i-1][::-1]:
                return len(s)-i-1
    return -1

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    q = int(input().strip())

    for q_itr in range(q):
        s = input()

        result = palindromeIndex(s)

        fptr.write(str(result) + '\n')

    fptr.close()

File: Algorithms/Strings/Palindrome Index.php
<?php

/*
 * Complete the 'palindromeIndex' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts STRING s as parameter.
 */

function palindromeIndex($s) {
    # Write your code here
    for ($i = 0; $i < strlen($s); $i++) {
        $temp = substr($s, 0, $i) . substr($s, $i + 1);
        if ($temp == strrev($temp)) {
            return $i;
        }
    }
    return -1;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$q = intval(trim(fgets(STDIN)));

for ($q_itr = 0; $q_itr < $q; $q_itr++) {
    $s = rtrim(fgets(STDIN), "\r\n");

    $result = palindromeIndex($s);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);
